web/controllers/survey_controller.ex

```diff
@@ -359,6 +360,69 @@ defmodule Ask.SurveyController do
     end
   end
 
+  def repeat(conn, %{"survey_id" => id}) do
```

Comment:
I would expect that repeat would call launch instead of duplicating the logic.

repeat should be

1. create new ocurrence
2. setup respondents
3. launch